tools/trace.py

```diff
@@ -299,6 +299,26 @@ def _generate_field_assign(self, idx):
                                         (idx, Probe.c_type[field_type], expr)
                 self._bail("unrecognized field type %s" % field_type)
 
+        def _generate_usdt_filter_read(self):
+            text = ""
+            if self.probe_type == "u":
+                    for arg, _ in Probe.aliases.items():
```

Comment:
Right, forgot about that. The filter is indeed quite unstructured. Good to go then.